blocks/library/index.js

```diff
@@ -1,25 +1,70 @@
-import './shortcode';
-import './image';
-import './gallery';
-import './heading';
-import './quote';
-import './embed';
-import './list';
-import './separator';
-import './more';
-import './button';
-import './pullquote';
-import './table';
-import './preformatted';
-import './code';
-import './html';
-import './freeform';
-import './latest-posts';
-import './categories';
-import './cover-image';
-import './text-columns';
-import './verse';
-import './video';
-import './audio';
-import './block';
-import './paragraph';
+/**
+ * Internal dependencies
+ */
+import {
+	registerBlockType,
+	setDefaultBlockName,
+	setUnknownTypeHandlerName,
+} from '../api';
+import * as audio from './audio';
+import * as button from './button';
+import * as categories from './categories';
+import * as code from './code';
+import * as coverImage from './cover-image';
+import * as embed from './embed';
+import * as freeform from './freeform';
+import * as gallery from './gallery';
+import * as heading from './heading';
+import * as html from './html';
+import * as image from './image';
+import * as latestPosts from './latest-posts';
+import * as list from './list';
+import * as more from './more';
+import * as paragraph from './paragraph';
+import * as preformatted from './preformatted';
+import * as pullquote from './pullquote';
+import * as quote from './quote';
+import * as reusableBlock from './block';
+import * as separator from './separator';
+import * as shortcode from './shortcode';
+import * as table from './table';
+import * as textColumns from './text-columns';
+import * as verse from './verse';
+import * as video from './video';
+
+export const registerCoreBlocks = () => {
+	[
+		audio,
+		button,
+		categories,
+		code,
+		coverImage,
+		embed,
+		...embed.common,
+		...embed.others,
+		freeform,
+		gallery,
+		heading,
+		html,
+		image,
+		list,
+		latestPosts,
+		more,
+		paragraph,
+		preformatted,
+		pullquote,
+		quote,
+		reusableBlock,
+		separator,
+		shortcode,
+		table,
+		textColumns,
+		verse,
+		video,
+	].forEach( ( { name, settings } ) => {
+		registerBlockType( name, settings );
+	} );
+
+	setDefaultBlockName( paragraph.name );
+	setUnknownTypeHandlerName( freeform.name );
+};
```

Comment:
I like this personally, I think it clarifies the config of the editor's block by centralizing them.
I'm not certain this solves all the "deferring issues" but it makes it clear when this initialization is happening (where this function is being called).

I'd also be open to refactor `registerBlockType` to move the name as part of the settings object like suggested by @aduth . But this could be a separate PR.

Reply:
Yes, I also like what @aduth proposed, but this might be too much for this PR. It's already big enough and we would have to update all docs to reflect this relatively small change. We would also have to promote it over the current API which can be found in a few example repositories, blog posts, online courses, etc :)